### PR TITLE
Generalized lemma in `tests/specs/lemmas.k`

### PR DESCRIPTION
**Pull request by Riib11**
In `tests/specs/mcd/verification.k` the lemma
```
rule #sizeByteArray(_WS [ START .. WIDTH ]) => WIDTH requires 0 <=Int START andBool 0 <=Int WIDTH [simplification]
```
is strictly more general that the following lemma from `tests/specs/lemmas.k`
```
rule #sizeByteArray(#buf(SIZE, _)[START .. WIDTH]) => WIDTH
  requires #range(0 <= START < SIZE)
   andBool #range(0 < WIDTH <= SIZE -Int START)
```
So, the more general lemma has been moved from `tests/specs/lemmas.k` to `tests/specs/mcd/verification.k`, and the less general lemma in `tests/specs/mcd/verification` has been removed. This works because the module in `tests/specs/mcd/verification.k` that used the less general lemma already imports the module in `tests/specs/lemmas.k` that now has the more general lemma.

### CODE REVIEW

**Comment by Riib11** [comment]
I think this fails checks for the same reason that #1005 fails checks. Will try again when the master branch is passing.

File: tests/specs/lemmas.k [hunk 1]
```diff
@@ -41,14 +41,12 @@ module LEMMAS
...
-       andBool #range(0 < WIDTH <= SIZE -Int START)                                                       [simplification]
-
-    rule #sizeByteArray(#range(_, START, WIDTH)) => WIDTH
-      requires WIDTH >=Int 0 andBool START >=Int 0                                                        [simplification]
+    rule #sizeByteArray(BUF1 ++ BUF2)                  => #sizeByteArray(BUF1) +Int #sizeByteArray(BUF2)     [simplification]
+    rule #sizeByteArray(#buf(SIZE, _))                 => SIZE                                               [simplification]
+    rule #sizeByteArray(_BUF [ START .. WIDTH ])       => WIDTH
+        requires 0 <=Int START andBool 0 <=Int WIDTH                                                         [simplification]
+    rule #sizeByteArray(#range(_, START, WIDTH))       => WIDTH
+      requires WIDTH >=Int 0 andBool START >=Int 0                                                           [simplification]
```
Comment (ehildenb) [inline tests/specs/lemmas.k:49]:
```suggestion
    rule #sizeByteArray(#range(_, START, WIDTH))       => WIDTH
      requires 0 <=Int WIDTH andBool 0 <=Int START                                                           [simplification]
```

**Comment by ehildenb** [comment]
This fails because the VMTests on the haskell backend are timing out on CI. I updated the branch, and if it passes tests I'll approve and merge (it may make it in the timeout this time).

I'm trying to remove the VMTests from CI on haskell backend (https://github.com/kframework/evm-semantics/pull/1042), but it's blocked on an upstream K issue.

File: tests/specs/lemmas.k [hunk 2]
```diff
@@ -41,14 +41,12 @@ module LEMMAS
...
-       andBool #range(0 < WIDTH <= SIZE -Int START)                                                       [simplification]
-
-    rule #sizeByteArray(#range(_, START, WIDTH)) => WIDTH
-      requires WIDTH >=Int 0 andBool START >=Int 0                                                        [simplification]
+    rule #sizeByteArray(BUF1 ++ BUF2)                  => #sizeByteArray(BUF1) +Int #sizeByteArray(BUF2)     [simplification]
+    rule #sizeByteArray(#buf(SIZE, _))                 => SIZE                                               [simplification]
+    rule #sizeByteArray(_BUF [ START .. WIDTH ])       => WIDTH
+        requires 0 <=Int START andBool 0 <=Int WIDTH                                                         [simplification]
+    rule #sizeByteArray(#range(_, START, WIDTH))       => WIDTH
+      requires 0 <=Int START andBool 0 <=Int WIDTH                                                           [simplification]
```
Comment (ehildenb) [inline tests/specs/lemmas.k:49]:
```suggestion
    rule #sizeByteArray(BUF1 ++ BUF2)            => #sizeByteArray(BUF1) +Int #sizeByteArray(BUF2)     [simplification]
    rule #sizeByteArray(#buf(SIZE, _))           => SIZE                                               [simplification]
    rule #sizeByteArray(_BUF [ START .. WIDTH ]) => WIDTH requires 0 <=Int START andBool 0 <=Int WIDTH [simplification]
    rule #sizeByteArray(#range(_, START, WIDTH)) => WIDTH requires 0 <=Int START andBool 0 <=Int WIDTH [simplification]
```

**Review by ehildenb — APPROVED** [review]

**Pull request merged**